tensorflow/c/eager/gradients.cc

```diff
@@ -263,7 +263,18 @@ Status AddInputList(AbstractOperation* op_,
 Status SetAttrString(AbstractOperation* op_, const char* attr_name,
                      const char* data, size_t length,
                      ForwardOperation* forward_op_) {
-  forward_op_->attrs.Set(attr_name, StringPiece(data, length));
+  std::cout << "inside SetAttrString:" << std::endl;
+  std::cout << "\tdata = " << data << ", len = " <<length << std::endl; 
+  std::cout << "\tattr_name = " << attr_name << std::endl;                     
+  forward_op_->attrs.Set(attr_name, StringPiece(data, length)); // <--this is not setting correctly
```

Comment:
@saxenasaurabh I'm logging the attributes right as they come in and right after we have set the attributes. 
The `attrs.Set()` for Strings and lists aren't working properly; for example here is the output log for the SetAttrString:

```
inside SetAttrString:
        data = SAME, len = 4
        attr_name = padding
        attrs.Get(padding) = 
exiting SetAttrString

Reply:
attrs.Get(padding) is returning empty string for some reason.

Reply:
I verified that this is a bug. I have a fix out for review.

Reply:
Could you check if a416b4ef0047a0a892cf429c1f41b391e3d5240c fixes this?

Reply:
That fixed it, thanks!